app/widgets/common/secrets/SecretActions.ts

```diff
@@ -48,23 +59,34 @@ export default class SecretActions {
         visibility: Visibility,
         hidden: boolean,
         providerName?: string,
-        providerOptions?: ProviderOptions | null
+        providerPath?: string
     ) {
-        return this.manager.doPut(`/secrets/${key}`, {
-            body: {
-                value,
-                visibility,
-                is_hidden_value: hidden,
+        let body: SecretRequestBody = {
+            key,
+            value,
+            visibility,
+            is_hidden_value: hidden
+        };
+        if (providerName && providerPath) {
+            body = {
+                ...body,
                 provider_name: providerName,
-                provider_options: providerOptions
-            }
-        });
+                provider_options: getSecretProviderOptions(providerPath)
+            };
+        }
+        return this.manager.doPut(`/secrets/${key}`, { body });
```

Comment:
Optional:

I guess you will be able to get better type security with:
```suggestion
        return this.manager.doPut<Secret, UpdateSecretRequestBody>(`/secrets/${key}`, { body });
```

Reply:
it should CreateSecretRequestBody not UpdateSecretRequestBody in that case :) https://github.com/cloudify-cosmo/cloudify-stage/pull/2427/commits/fed30641b28375a29e8da15227b22f287c0c8e0a - done ✔️